homeassistant/components/mill/__init__.py

```diff
@@ -55,6 +56,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     await data_coordinator.async_config_entry_first_refresh()
     hass.data[DOMAIN][conn_type][key] = data_coordinator
 
+    if entry.data.get(CONNECTION_TYPE) == CLOUD:
+        device_registry = dr.async_get(hass)
+        for mill_device in mill_data_connection.devices.values():
+            # migrate to new device ids
+            device_entry = device_registry.async_get_device(
+                identifiers={(DOMAIN, mill_device.device_id)}
+            )
+            if device_entry and entry.entry_id in device_entry.config_entries:
+                device_registry.async_update_device(
+                    device_entry.id, new_identifiers={(DOMAIN, mill_device.mac_address)}
+                )
```

Comment:
Why don't we add them as connections?

Reply:
Not so familiar with that.

So `connections={(CONNECTION_NETWORK_MAC, mac)},` instead of changing the `identifiers ` ?
Will that both local and cloud connected devices?

Reply:
devices can have both identifiers and connections, so we can keep the current connections and just make sure they have the same connections

Reply:
Was this what you meant? https://github.com/home-assistant/core/pull/133510/commits/a1bfc46e6cfa351614952888c942fd002c536097

Will need to test it tomorrow